katportalclient/test/test_client.py

```diff
@@ -921,36 +968,118 @@ def test_sensor_value_one_result(self):
+                                                         include_value_time=True)
+        expected_result = {
+            "anc_tfr_m018_l_band_offset": SensorSampleValueTime(sample_time=1531302437,
+                                                              value_time=1530713112,
+                                                              value=43680.0,
+                                                              status='nominal'),
+            "some_other_sample": SensorSampleValueTime(sample_time=222.222,
+                                                     value_time=111.111,
+                                                     value=43680.0,
+                                                     status='nominal')}
+        assert result == expected_result
+
+    @gen_test
+    def test_sensor_values_multiple_filters_multiple_matches(self):
+        """Test that we handle multiple matches correctly with multiple filters"""
+
+        mon_response_0 = ('[{"status":"nominal",'
+                          '"name":"some_other_sample_0",'
+                          '"component":"anc","value":43480.0,'
+                          '"value_time":110.111,"sample_time":220.222}]')
+        mon_response_1 = ('[{"status":"nominal",'
+                          '"name":"some_other_sample_1",'
+                          '"component":"anc","value":43580.0,'
+                          '"value_time":111.111,"sample_time":221.222}]')
+
+        self.mock_http_async_client().fetch.side_effect = self.mock_async_fetchers(
+            valid_responses=[mon_response_0, mon_response_1],
+            invalid_responses=['1error', '2error'],
+            containses=["sample_0", "sample_1"]
+            )
+
+        expected_result = {"some_other_sample_0": SensorSample(timestamp=220.222,
+                                                               value=43480.0,
+                                                               status='nominal'),
+                           "some_other_sample_1": SensorSample(timestamp=221.222,
+                                                               value=43580.0,
+                                                               status='nominal')}
+        res = yield self._portal_client.sensor_values(["some_other_sample_0",
+                                                       "some_other_sample_1"])
+        assert res == expected_result, res
+
     @gen_test
     def test_sensor_history_single_sensor_without_value_time(self):
-        """Test time ordered data without value_time for single sensor request."""
+        """Test that time ordered data with value_time is received for a
+        single sensor request.
+        """
         history_base_url = self._portal_client.sitemap[
             'historic_sensor_values']
         sensor_name = 'anc_mean_wind_speed'
         publish_messages = [sensor_history_pub_messages_json['init']]
         publish_messages.extend(sensor_history_pub_messages_json[sensor_name])
```

Comment:
All the `publish_messages` parts can be removed (this test and the ones that follow).  They are left over from old katstore tests.
The `sensor_history_pub_messages_json` variable must be deleted too.